sal-common/src/main/java/org/ow2/proactive/sal/model/Deployment.java

```diff
@@ -103,6 +108,32 @@ public Node getNode() {
         }
     }
 
+    public static Deployment fromId(String nodeName) {
+        Deployment deployment = null;
+        ApplicationContext applicationContext = SpringConfiguration.contextProvider().getApplicationContext();
+        if (applicationContext != null) {
+            DeploymentRepository deploymentRepository = (DeploymentRepository) applicationContext.getBean("deploymentRepository");
+            deployment = deploymentRepository.findOne(nodeName);
+        }
+        if (deployment == null) {
+            deployment = new Deployment();
+            deployment.nodeName = nodeName;
+        }
+        return deployment;
+    }
+
+    @JsonSetter("cloudId")
+    public void setPaCloudById(String cloudId) {
+        this.paCloud = PACloud.fromId(cloudId);
+        this.paCloud.addDeployment(this);
+    }
+
+    @JsonSetter("taskId")
+    public void setTaskById(String taskId) {
+        this.task = Task.fromId(taskId);
+        this.task.addDeployment(this);
+    }
```

Comment:
In this case, this will be done manually in the DeploymentService class instead of here with using `@JsonSetter`

Reply:
Same here. This is needed for deserialisation.